client/src/pages/admin/CampaignsTable/index.jsx

```diff
@@ -0,0 +1,257 @@
+/* eslint-disable no-unused-vars */
+import React, { useEffect, useState } from 'react';
+import {
+  Collapse,
+  Table,
+  Popconfirm,
+  Popover,
+  Space,
+  message,
+  Pagination,
+} from 'antd';
+import {
+  CloseCircleOutlined,
+  DeleteOutlined,
+  FileSearchOutlined,
+  UserAddOutlined,
+} from '@ant-design/icons';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+import './style.css';
+
+function CampaignsTable() {
+  const [campaigns, setCampaigns] = useState([]);
+  const [page, setPage] = useState(1);
+  const [campaignsCount, setCampaignsCount] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    const source = axios.CancelToken.source();
+    const getCampaigns = async () => {
+      try {
+        const {
+          data: {
+            data: { campaigns: campaignsData, count },
+          },
+        } = await axios.get(`/api/admin/campaigns?page=${page}`, {
+          cancelToken: source.token,
+        });
+        setCampaigns(campaignsData);
+        setCampaignsCount(count);
+        setIsLoading(false);
+        setIsError(false);
+      } catch ({
+        response: {
+          data: { message: errorMessage },
+        },
+      }) {
+        message.error(errorMessage);
+        setIsError(true);
+      }
+    };
+    getCampaigns();
+    return () => {
+      source.cancel();
+    };
+  }, [page]);
+  const handleDeleteCampaign = async (id) => {
+    setCampaigns(campaigns.filter((campaign) => campaign.id !== id));
```

Comment:
I think this line should be in try after await Axios that because if there is error in delete in back not delete in front